### PR TITLE
Prevent update fail and alert on error

### PR DESCRIPTION
**Pull request by Ritesh-patel**
- Disable Post Update button if there is no content in editor or content is too large
- On failed update, show error message and prevent editor from closing

### CODE REVIEW

**Comment by Ritesh-patel** [comment]
@adamsilverstein can you take a look at this one? I have merged "Disable post update" and "Failure alerts" in single PR.

File: src/react/components/TinyMCEEditor.js [hunk 1]
```diff
@@ -44,14 +53,39 @@ class TinyMCEEditor extends Component {
...
 
+  componentDidUpdate() {
+    const thisError = this.props.editorContainer.props.api.error;
+    /*
+    could have used prevProps here to check error in previous props
+    but somehow both previous and current props are same
+    Using `isError` for current editor, we can prevent infinite loop
+     */
+    if (thisError && tinymce.activeEditor.setError && !tinymce.activeEditor.isError) {
+      tinymce.activeEditor.setError(true, this.props.editorContainer.props.api.errorMessage);
```
Comment (adamsilverstein) [inline src/react/components/TinyMCEEditor.js]: `this.props.editorContainer.props.api.errorMessage` seems overly verbose, can you pass as props directly to the `TinyMCEEditor` component?

File: src/react/components/TinyMCEEditor.js [hunk 2]
```diff
@@ -44,14 +53,39 @@ class TinyMCEEditor extends Component {
...
+          setPostingEnable();
+        }, 250));
+        setPostingEnable();
         tinymce.activeEditor.focus(); // Set focus to active editor
       }, 250);
     }, 10);
   }
 
+  componentDidUpdate() {
+    const thisError = this.props.editorContainer.props.api.error;
```
Comment (adamsilverstein) [inline src/react/components/TinyMCEEditor.js]: can you pass `this.props.editorContainer.props.api.error` directly as props to the `TinyMCEEditor` component?

File: src/react/components/TinyMCEEditor.js [hunk 3]
```diff
@@ -44,14 +53,39 @@ class TinyMCEEditor extends Component {
         const stateContent = this.props.rawText;
         tinymce.activeEditor.clearAuthors = this.props.clearAuthors;
         tinymce.activeEditor.clearHeadline = this.props.clearHeadline;
+        tinymce.activeEditor.setPostingEnable = this.props.setPostingEnable;
```
Comment (adamsilverstein) [inline src/react/components/TinyMCEEditor.js]: lets clean up this and the three previous lines (and anything you need to get from props) with destructuring:

const {
clearAuthors,
clearHeadline,
setPostingEnable,
setError, 
} = this.props;

File: src/react/components/TinyMCEEditor.js [hunk 4]
```diff
@@ -44,14 +53,39 @@ class TinyMCEEditor extends Component {
...
+    /*
+    could have used prevProps here to check error in previous props
+    but somehow both previous and current props are same
+    Using `isError` for current editor, we can prevent infinite loop
+     */
+    if (thisError && tinymce.activeEditor.setError && !tinymce.activeEditor.isError) {
+      tinymce.activeEditor.setError(true, this.props.editorContainer.props.api.errorMessage);
+      tinymce.activeEditor.isError = true;
+      setTimeout(() => {
+        tinymce.activeEditor.setError(false, this.props.editorContainer.props.api.errorMessage);
```
Comment (adamsilverstein) [inline src/react/components/TinyMCEEditor.js]: replace `this.props.editorContainer.props.api.errorMessage` with props passed to component

File: src/react/components/TinyMCEEditor.js (hunk 3 above)
Comment (Ritesh-patel) [reply src/react/components/TinyMCEEditor.js]: Understood but problem is we have same methods defined in upper scope https://github.com/Ritesh-patel/liveblog/blob/89fb6d67c39e05ac70225f4ef05aa3e792fa97be/src/react/components/TinyMCEEditor.js#L25, we can change names though but then we will need to change it into other places as well.
We can do something like this though
```
const editorMethods = {
          clearAuthors: this.props.clearAuthors,
          clearHeadline: this.props.clearHeadline,
          setPostingEnable: this.props.setPostingEnable,
          setError: this.props.setError,
};
tinymce.activeEditor = Object.assign(tinymce.activeEditor, editorMethods);
```

Comment (Ritesh-patel) [reply src/react/components/TinyMCEEditor.js]: No `Object.assign` doesn't work.

**Review by adamsilverstein — APPROVED** [review]
LGTM 👍

**Comment by Ritesh-patel** [comment]
@paulschreiber this is ready now.

File: src/react/Editor/Editor.js [hunk 5]
```diff
@@ -328,6 +328,9 @@ class EditorWrapper extends Component {
       clearAuthors,
       clearHeadline,
       rawText,
+      setPostingEnable,
```
Comment (paulschreiber) [inline src/react/Editor/Editor.js]: Should this be `setPostingEnabled`?

File: src/react/components/TinyMCEEditor.js [hunk 6]
```diff
@@ -33,6 +34,14 @@ export const clearHeadline = () => {
   }
 };
 
+export const setPostingEnable = () => {
+  if (tinymce.activeEditor.setPostingEnable) {
+    const content = tinymce.activeEditor.getContent();
+    const postingEnabled = content.length > 0 && content.length < 65535;
```
Comment (paulschreiber) [inline src/react/components/TinyMCEEditor.js]: Put 65535 in a constant.

File: src/react/components/TinyMCEEditor.js (hunk 3 above)
Comment (paulschreiber) [reply src/react/components/TinyMCEEditor.js]: @adamsilverstein is this ok?

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: src/react/Editor/Editor.js (hunk 5 above)
Comment (Ritesh-patel) [reply src/react/Editor/Editor.js]: It's a function which can take true/false value.`setPostingEnabled` is more like for setting a flag.

Comment (paulschreiber) [reply src/react/Editor/Editor.js]: This is ungrammatical in its current form. You could go with setEnablePosting if Enable is to be treated as a verb.

Comment (Ritesh-patel) [reply src/react/Editor/Editor.js]: Agreed, setEnablePosting sounds better.

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**